packages/aws-rfdk/lib/deadline/lib/worker-configuration.ts

```diff
@@ -29,6 +29,39 @@ import {
   Version,
 } from './version';
 
+/**
+ * Provider for adding user data scripts
+ * Methods of this class will be invoked in WorkerInstanceConfiguration
+ * on different stages of worker configuration
+ */
+export interface IInstanceUserDataProvider {
+  // Method that is invoked before configuration Cloud Watch Agent.
+  preCloudWatchAgent(host: IHost): void;
+  // Method that is invoked before RenderQueue configuration.
+  preRenderQueueConfiguration(host: IHost): void;
+  // Method that is invoked before Deadline worker configuration.
+  preDeadlineConfiguration(host: IHost): void;
+  // Method that is invoked after all configuration is done and worker started.
+  postWorkerLaunch(host: IHost): void;
+}
+
+export class InstanceUserDataProvider extends Construct  implements IInstanceUserDataProvider{
```

Comment:
Missing a docstring on this class.

Reply:
I think this needs more documentation. This class is the API that RFDK users should use in order to achieve custom user data. In here, we should explain that users can sub-class this and override the desired methods to add custom user data commands for `WorkerInstanceFleet` or `WorkerInstanceConfiguration`.